### ISSUE TITLE
How to remove `v_num` from the progress bar ?

### ISSUE THREAD

**Issue by phihung** [PyTorchLightning/pytorch-lightning#1595]
Version: 0.7.3

The `v_num` is automatically added to progress_bar when some logger is used
It is not much a problem for tensorboard when v_num is just a simple number
But v_num for mlfow takes a lot of space

Traning step
```
def training_step(self, batch, batch_nb):
    ....
    log = { "trn_loss": 0.1, "lr": 0.001 }
    return {"loss": loss, "log": log, "progress_bar": log}
```

Progress bar when using with mlflow logger
```
 [00:33<00:46, 1.62s/it, loss=0.740, lr=8e-6, trn_loss=0.659, v_num=18_28bc973b1f0e42e8b4d664d1ef7812f6]
```

Also, `loss` is automatically added to `progress_bar`

**Comment by williamFalcon** [comment]
loss and trn_loss are the same. you’re duplicating work.

**Comment by williamFalcon** [comment]
maybe we can show only the last k letters for other exps

**Comment by phihung** [comment]
It may be a good default to show loss and v_num in progress bar when no `progress_bar` property in the return dict. 
But here I explicitly ask to show only my trn_loss and lr (whatever they are). Why does package add loss and v_num ?

**Comment by williamFalcon** [comment]
normally we run many (hundreds) of experiments to find a good set of hyperparams or a model that works for our task. as such, it’s very useful to know which experiment version you are running - v_num (experiment version number)

**Comment by phihung** [comment]
Thanks a lot @williamFalcon  for the explanation. 
> maybe we can show only the last k letters for other exps
That should work.

**Comment by DKandrew** [comment]
Side question: is the meaning of v_num being documented anywhere in the introduction guide? It looks very mysterious to me until I read this post.

**Comment by Borda** [comment]
@DKandrew mind share your proposal how would you describe in a PR?

**Comment by DKandrew** [comment]
@Borda Do you mean where to introduce `v_num`? I think a good place would be https://pytorch-lightning.readthedocs.io/en/latest/introduction_guide.html the "Train on CPU" section. That is the first time when we(beginner) encounter the progress bar.

**Comment by radao** [comment]
Little bit of a hack but you can remove it manually by adding the following method in your lightning module:
```
def get_progress_bar_dict(self):
        tqdm_dict = super().get_progress_bar_dict()
        if 'v_num' in tqdm_dict:
            del tqdm_dict['v_num']
        return tqdm_dict
```

**Comment by nagyrajmund** [comment]
I also had no idea what v_num was until I read William's comment above. 

Would renaming it to `version` be a solution?

Also, if it comes from the logger's `version` attribute then that's another reason to rename it (from `v_num` to `version`) since that isn't necessarily a number.

**Comment by awaelchli** [comment]
I created a PR here #2594 that truncates the long version numbers to 3 digits and also added docs how to override the defaults. 
What @radao describes as a hack I would say is actually the recommended way. And it is not a hack because like any other hook in the LightningModule, you are free to override anything you want. For example, to drop the v_num completely, you would do

```python
def get_progress_bar_dict(self):
    tqdm_dict = super().get_progress_bar_dict()
    tqdm_dict.pop("v_num", None)
    return tqdm_dict
```
The same approach can be used to rename it to "version" @nagyrajmund

**Comment by awaelchli** [comment]
Which length (num characters) is acceptable to truncate to? 
- 3 (currently)
- ...
- 7 (same length as git commit)
- any other suggestions?

**Comment by nagyrajmund** [comment]
I think truncating to 3 is appropriate for both use cases (i.e. strings and simple numbers).

**Comment by xujiaze13** [comment]
What is `v_num` meaning?

**Comment by awaelchli** [comment]
@xujiaze13 v_num is the version number of the experiment. Depending on the logger, this can be a simple counter that increases everytime you run the trainer. For other loggers, they have alphanumeric identifiers which can be quite long and fill up much of the tqdm progress bar, leaving no room for the important metrics to monitor.

**Comment by jxu124** [comment]
Can it still be like this? I don't see any key descriptions about this return dict in the document. 
https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.core.LightningModule.html#lightning.pytorch.core.LightningModule.training_step

**Comment by 2catycm** [comment]
It is mysterious when there are multiple logger. For examaple, I use both  TensorboardLogger and CSVLogger, it seems one takes the "lightning_logs/version_53" directory and one takes the "lightning_logs/version_54" directory. And the shown `v_num` in progress bar is "3_54". The meaning of "3_" is not clear to me.

**Comment by 2catycm** [comment]
And it is not a changing variable during training progress, may be `v_num` should be printed at the begining of the training to notice the user about which version is running, and where will be the logs reside, instead of showing in the progress bar.

**Comment by bpawlik** [comment]
I solved it with:
```
from lightning.pytorch.callbacks import TQDMProgressBar

class LitProgressBar(TQDMProgressBar):
    def get_metrics(self, trainer, model):
        # don't show the version number
        items = super().get_metrics(trainer, model)
        items.pop("v_num", None)
        return items
```

and then:
```
trainer = L.Trainer(
        callbacks=[LitProgressBar(leave=False)]
    )
```
